src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+
+### fluid.flatten(array)
+
+Converts an array consisting of a mixture of arrays and non-arrays into the concatenation of any inner arrays 
+with the non-array elements. The original array *** will not be modified ***. See description of [mapcat](http://martinfowler.com/articles/collection-pipeline/flat-map.html) or `flat-map`.
+
+* `array {Array}` The array to be flattened
+* Returns: `{Array}` The flattened array.
+
+
+## Sorting and searching
+
+### fluid.find(source, func[, deflt])
+
+Scan through a list or hash of objects, terminating on the first member which
+matches a predicate function. The return is the return value from the predicate.
+* `source {Arrayable|Object}` The list or hash of objects to be searched.
+* `func {Function: (element {Any}, index {Number|String}) → Any}` A predicate function, acting on a member. A predicate which
+returns any value which is not `undefined` will terminate the search.
+* `deflt {Object|Undefined}` [optional] A value to be returned in the case no predicate function matches
+a member. The default will be the natural value of `undefined`
+* Returns: The first return value from the predicate function which is not `undefined`
+
+### fluid.find_if(source, func[, deflt])
+
+Scan through a list or hash of objects, terminating on the first member which
+matches a predicate function. The return value is the first element itself for which the predicate returns other than `false`.
+* `source {Arrayable|Object}` The list or hash of objects to be searched.
+* `func {Function: (element {Any}, index {Number}) → Boolean}` A predicate function, acting on a member. A predicate which
+returns any value which is not `false` will terminate the search.
+* `deflt {Object}` [optional] A value to be returned in the case no predicate function matches a
+member. The default will be the natural value of `undefined`
+* Returns: The first element for which the value of the predicate function is not `false`
+
+### fluid.contains(tosearch, value)
+
+Searches through the supplied structure, and returns `true` if the supplied value can be found, using `===` equality
+
+* `tosearch {Object|Array}` The structure to be searched for the supplied value,
+* `value {Any}` The value to be searched for
+* Returns: `{Boolean|Undefined}` Returns `true` if the value was found in the container
+
+### fluid.keyForValue(tosearch, value) {
+
+Searches through the supplied object for the first value which matches the one supplied.
+
+* `tosearch {Object|Array}` The structure to be searched for the supplied value
+* `value {Any}` The value to be found. This will be compared against the structure's members using `===` equality.
+* Returns: `{String|Integer|Undefined}` The first key whose value matches the one supplied, or `undefined` is no value matches
```

Comment:
Also, it would probably be good to be explicit about the fact that if tosearch is an array, the return value is the index of the found item